web/src/App.tsx

```diff
@@ -9,29 +9,44 @@ import Logout from "./screens/auth/logout";
 import { baseUrl } from "./utils";
 
 import { AuthContext, SettingsContext } from "./utils/Context";
+import {Fragment, useEffect} from "react";
+import {APIClient} from "./api/APIClient";
+import {useCookies} from "react-cookie";
 
 function Protected() {
+    const [authContext, setAuthContext] = AuthContext.use();
```

Comment:
I'd say that all the changes made in this file are unneeded. I've tested the solution proposed in my two comments from below and it works as expected -- on an invalid cookie, upon the first next network activity, the user gets logged out and the cookie gets cleared.